File: output/GPT4o-Mini/xCodeEval_4000/C++/C/40ca10a54463a16f6c0559d2d974fdc5.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <math.h>

const int N = 18000;
const int M = 18000;
bool is[N]; 
int prm[M], cntp;

int getprm(int n) {
    int i, j, k = 0;
    int s, e = (int)(sqrt(0.0 + n) + 1);
    memset(is, 1, sizeof(is));
    prm[k++] = 2; 
    is[0] = is[1] = 0;
    for (i = 4; i < n; i += 2) is[i] = 0;
    for (i = 3; i < e; i += 2) 
        if (is[i]) {
            prm[k++] = i;
            for (s = i * 2, j = i * i; j < n; j += s)
                is[j] = 0;
        }
    for (; i < n; i += 2) 
        if (is[i]) prm[k++] = i;
    return k; 
}

File: output/GPT4o-Mini/xCodeEval_4000/C++/Go/40ca10a54463a16f6c0559d2d974fdc5.go
package main

import (
	"fmt"
	"math"
)

const N = 18000
const M = 18000

var is [N]bool
var prm [M]int
var cntp int

func getprm(n int) int {
	var k int
	e := int(math.Sqrt(float64(n)) + 1)
	for i := range is {
		is[i] = true
	}
	prm[k] = 2
	k++
	is[0] = false
	is[1] = false
	for i := 4; i < n; i += 2 {
		is[i] = false
	}
	for i := 3; i < e; i += 2 {
		if is[i] {
			prm[k] = i
			k++
			for s, j := i*2, i*i; j < n; j += s {
				is[j] = false
			}
		}
	}
	for i := e; i < n; i += 2 {
		if is[i] {
			prm[k] = i
			k++
		}
	}
	return k
}

